packages/sync/src/utils/countSyncSnapshotRecords.ts

```diff
@@ -0,0 +1,26 @@
+import { TupaiaDatabase } from '@tupaia/database';
+
+import { RecordType, SyncSessionDirectionValues } from '../types';
+import { getSnapshotTableName } from './manageSnapshotTable';
+
+export const countSyncSnapshotRecords = async (
+  database: TupaiaDatabase,
+  sessionId: string,
+  direction: SyncSessionDirectionValues,
+  recordType?: RecordType,
+): Promise<number> => {
+  const tableName = getSnapshotTableName(sessionId);
+
+  const [{ total }] = (await database.executeSql(
+    `
+      SELECT count(*) AS total FROM ${tableName}
+      WHERE direction = :direction
```

Comment:
Can we cut out the concept of direction, if we are using a different structure as the holding space for incoming records compared to snapshot of outgoing?

Reply:
We're only using different structure for incoming records for client, and we still need both for the central server, so I think we should still keep this one